docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,201 @@
+
+    2. `LTTng`: Used to capture event data emitted at runtime by CoreCLR. This data is then used to analyze the behavior of various runtime components such as the GC, JIT and thread pool.
+
+## Collecting a Trace
+
+1. Have two shells available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
+
+2. **[App]** Setup the application shell with the following environment variables - this enables tracing configuration of CoreCLR.
+
+    > ```bash
+    > export COMPlus_PerfMapEnabled=1
+    > export COMPlus_EnableEventLog=1
+    > ```
+
+3. **[Trace]** Start collection.
+
+    > ```bash
+    > sudo ./perfcollect collect sampleTrace
+    > ```
+
+    Expected Output:
+
+    > ```bash
+    > Collection started.  Press CTRL+C to stop.
+    > ```
+
+4. **[App]** Run the app - let it run as long as you need to in order to capture the performance problem.
+
+    > ```bash
+    > dotnet run
+    > ```
+
+5. **[Trace]** Stop collection - hit CTRL+C.
+
+    > ```bash
+    > ^C
+    > ...STOPPED.
+    >
+    > Starting post-processing. This may take some time.
+    >
+    > Generating native image symbol files
+    > ...SKIPPED
+    > Saving native symbols
+    > ...FINISHED
+    > Exporting perf.data file
+    > ...FINISHED
+    > Compressing trace files
+    > ...FINISHED
+    > Cleaning up artifacts
+    > ...FINISHED
+    >
+    > Trace saved to sampleTrace.trace.zip
+    > ```
+
+    The compressed trace file is now stored in the current working directory.
+
+## Resolving Framework Symbols
+
+Framework symbols need to be manually generated at the time the trace is collected. They are different than app-level symbols because the framework is pre-compiled while apps are just-in-time-compiled. For code like the framework that was precompiled to native code, you need a special tool called crossgen that knows how to generate the mapping from the native code to the name of the methods.
+
+`perfcollect` can handle most of the details for you, but it needs to have the crossgen tool and by default this is not part of the standard .NET distribution. If it is not there it warns you and refers you to these instructions. To fix things you need to fetch exactly the right version of crossgen for the runtime you happen to be using. If you place the crossgen tool in the same directory as the .NET Runtime DLLs (e.g. libcoreclr.so), then `perfcollect` can find it and add the framework symbols to the trace file for you.
+
+Normally when you create a .NET application, it just generates the DLL for the code you wrote, using a shared copy of the runtime for the rest.   However you can also generate what is called a 'self-contained' version of an application and this contains all runtime DLLs. The crossgen tool is part of the Nuget package that is used to create self-contained apps, so one way of getting the right crossgen tool is to create a self-contained package of your application.
```

Comment:
Do we refer to crossgen as "the crossgen tool" elsewhere? It sounds awkward to me.